fast64_internal/oot/scene/exporter/to_c/scene_cutscene.py

```diff
@@ -1,144 +1,95 @@
-from .....utility import CData, PluginError
+from .....utility import CData, PluginError, indent
 from ....oot_level_classes import OOTScene
 from ....cutscene.constants import ootEnumCSTextboxTypeEntryC, ootEnumCSListTypeListC, ootEnumCSListTypeEntryC
+from ....cutscene.exporter import OOTCutscene
 
 
-def ootCutsceneDataToC(csParent, csName):
-    # csParent can be OOTCutscene or OOTScene
-    data = CData()
-    data.header = "extern CutsceneData " + csName + "[];\n"
-    data.source = "CutsceneData " + csName + "[] = {\n"
-    nentries = len(csParent.csLists) + (1 if csParent.csWriteTerminator else 0)
-    data.source += "\tCS_BEGIN_CUTSCENE(" + str(nentries) + ", " + str(csParent.csEndFrame) + "),\n"
-    if csParent.csWriteTerminator:
-        data.source += (
-            "\tCS_TERMINATOR("
-            + str(csParent.csTermIdx)
-            + ", "
-            + str(csParent.csTermStart)
-            + ", "
-            + str(csParent.csTermEnd)
-            + "),\n"
+def ootCutsceneDataToC(csParent: OOTCutscene | OOTScene, csName: str):
+    csData = CData()
+    arrayName = f"CutsceneData {csName}[]"
```

Comment:
imo the name of the array is just `csName`, I would call this `declarationBase` since it's used for the extern declaration and the definition (which is also a declaration)

(declaration: "this thing exists"
definition: "this thing is this")

Reply:
names aren't my strongest point ngl lol

I'll change other names like that too to stay consistent 👍